pkg/server/test/list_objects.go

```diff
@@ -308,132 +334,88 @@ func TestListObjectsRespectsMaxResults(t *testing.T, ds storage.OpenFGADatastore
-			Relation:             "viewer",
-			User:                 "user:999",
-		})
-	}
-
-	listObjectsResponse = r
+	return model, modelID, numberObjectsAccesible
 }
 
-func BenchmarkListObjectsWithConcurrentChecks(b *testing.B, ds storage.OpenFGADatastore) {
+func BenchmarkListObjects(b *testing.B, ds storage.OpenFGADatastore) {
 	ctx := context.Background()
 	store := ulid.Make().String()
 
-	typedefs := parser.MustParse(`
-	type user
-
-	type document
-	  relations
-	    define allowed: [user] as self
-	    define viewer: [user] as self and allowed
-	`)
+	model, modelID, numberObjectsAccessible := setupListObjectsBenchmark(b, ds, store)
+	ctx = typesystem.ContextWithTypesystem(ctx, typesystem.New(model))
 
-	model := &openfgav1.AuthorizationModel{
-		Id:              ulid.Make().String(),
-		SchemaVersion:   typesystem.SchemaVersion1_1,
-		TypeDefinitions: typedefs,
+	req := &openfgav1.ListObjectsRequest{
+		StoreId:              store,
+		AuthorizationModelId: modelID,
+		Type:                 "document",
+		Relation:             "viewer",
+		User:                 "user:maria",
 	}
-	err := ds.WriteAuthorizationModel(ctx, store, model)
-	require.NoError(b, err)
 
-	n := 0
-	for i := 0; i < 100; i++ {
-		var tuples []*openfgav1.TupleKey
+	b.ResetTimer()
 
-		for j := 0; j < ds.MaxTuplesPerWrite()/2; j++ {
-			obj := fmt.Sprintf("document:%s", strconv.Itoa(n))
-			user := fmt.Sprintf("user:%s", strconv.Itoa(n))
+	var r *commands.ListObjectsResponse
 
-			tuples = append(
-				tuples,
-				tuple.NewTupleKey(obj, "viewer", user),
-				tuple.NewTupleKey(obj, "allowed", user),
-			)
+	var oneResultIterations, allResultsIterations int
 
-			n += 1
+	b.Run("oneResult", func(b *testing.B) {
+		listObjectsQuery := commands.NewListObjectsQuery(ds,
+			commands.WithListObjectsMaxResults(1),
+		)
+		for i := 0; i < b.N; i++ {
+			r, _ := listObjectsQuery.Execute(ctx, req)
+			require.Len(b, r.Objects, 1)
 		}
 
-		err = ds.Write(ctx, store, nil, tuples)
-		require.NoError(b, err)
-	}
-
-	listObjectsQuery := commands.NewListObjectsQuery(ds)
-
-	var r *commands.ListObjectsResponse
-
-	ctx = typesystem.ContextWithTypesystem(ctx, typesystem.New(model))
+		listObjectsResponse = r
+		oneResultIterations = b.N
+	})
+	b.Run("allResults", func(b *testing.B) {
+		listObjectsQuery := commands.NewListObjectsQuery(ds,
+			commands.WithListObjectsMaxResults(0),
+		)
+		for i := 0; i < b.N; i++ {
+			r, _ := listObjectsQuery.Execute(ctx, req)
+			require.Len(b, r.Objects, numberObjectsAccessible)
+		}
 
-	b.ResetTimer()
-	for i := 0; i < b.N; i++ {
-		r, _ = listObjectsQuery.Execute(ctx, &openfgav1.ListObjectsRequest{
-			StoreId:              store,
-			AuthorizationModelId: model.Id,
-			Type:                 "document",
-			Relation:             "viewer",
-			User:                 "user:999",
-		})
-	}
+		listObjectsResponse = r
+		allResultsIterations = b.N
+	})
 
-	listObjectsResponse = r
+	require.Greater(b, oneResultIterations, allResultsIterations)
```

Comment:
Not comparing `b.Elapsed()` here because `benchtime=5s` for all benchmarks, so the comparison is useless and sometimes misleading